### ISSUE TITLE
VS Code extension marks code unreachable when not needed (presumably)

### ISSUE THREAD

**Issue by BIDMAL** [microsoft/pyright#317]
**Describe the bug**
VS Code extension marks code unreachable when not needed (presumably)

**To Reproduce**
Input code from Code section

**Expected behavior**
No unreachable lines

**Screenshots or Code**
```
import socket

with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
    sock.bind(("", 10002))
    sock.listen()
    conn, addr = sock.accept()
    while True:        
        conn.settimeout(5)
        with conn:
            while True:
                try:
                    data = conn.recv(1024)
                except socket.timeout:
                    print("connection closed by timeout")
                    break

                if not data:
                    break
                print(data.decode("utf8"))
```

**VS Code extension or command-line**
VS Code extension 1.0.81

**Additional context**

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release of pyright.

**Comment by erictraut** [comment]
This is now fixed in 1.0.82, which I just published.
